doc/src/build/dependency-overrides.md

```diff
@@ -0,0 +1,57 @@
+---
+title: Dependency Overrides
+---
+
+One of the main features of the Move programming language used to program Sui smart contracts is code reuse. Packages implementing commonly used functionality, such as various cryptographic algorithms, can be simply reused by other developers that require this kind of functionality in their projects. As part of the package's lifecycle, its developers have the ability to [upgrade](./package-upgrades.md) a package to a newer version (e.g., to fix bugs or provide additional functionality), possibly multiple times. This naturally leads to a situation where some "user" packages may depend on one version of a "library" package while other "user" packages may depend on another. 
+
+A package might depend on another package _directly_ or _indirectly_ (also called _transitively_). A direct dependency is specified in the "user" package's [manifest](./move/manifest.md) file and its version is directly controlled by the developer of the "user" package. Clearly these directly dependent packages might have dependencies of their own, and these dependencies become indirect dependencies of the "user" package if they are not also specified in the "user" package's manifest file.
+
+
+As long as all versions of all (directly or indirectly) dependent packages are the same, no further action on the side of the "user" package's developer is required. Unfortunately, this is not always the case. Consider the following example of the `user` package directly depending on two "library" packages: `vault` and `currency` (these are not real packages and their addresses do not work). The (simplified) manifest file of the `user` package would then resemble the following:
+
+```move
+[package]
+name = "user"
+version = "1.0.0"
+
+[dependencies]
+vault = { git = "https://github.com/vault_org/vault.git" }
+currency = { git = "https://github.com/currency_org/currency.git" }
+```
+
+Further consider that both `vault` and `currency` depend on another "library" package, `crypto`, but each of them depends on a different version (`crypto` becomes an indirect dependency of the `user` package). Their respective manifest files could then resemble the following:
+
+```move
+[package]
+name = "vault"
+version = "1.0.0"
+
+[dependencies]
+crypto = { git = "https://github.com/crypto.org/crypto.git" , rev = "v1.0.0"}
+```
+
+```move
+[package]
+name = "currency"
+version = "2.0.0"
+
+[dependencies]
+crypto = { git = "https://github.com/crypto.org/crypto.git" , rev = "v2.0.0"}
+```
+
+This situation represents the [_diamond dependency conflict_](https://jlbp.dev/what-is-a-diamond-dependency-conflict) where two indirect dependencies in turn depend on different versions of the same package. If all packages involved are developed independently, then the developer of the `user` package has a problem -- the Move compiler cannot discern which version of the `crypto` package to use and, as a result, building of the `user` package fails.
```

Comment:
> https://jlbp.dev/what-is-a-diamond-dependency-conflict

I think I picked this link originally when I was writing our internal design document, but this may not be suitable for our public-facing docs.  Thoughts @ronny-mysten / @randall-Mysten ?  I think it's fine to drop the link entirely, to be honest, because we're explaining what it is below.